docs/current/guides/421437-work-with-host-filesystem.md

```diff
@@ -212,6 +212,28 @@ The following example creates a file in a container's `/tmp` directory and then
 </TabItem>
 </Tabs>
 
+## Important notes
+
+Using the host filesystem in your Dagger pipeline is convenient, but there are some important considerations to keep in mind:
+
+- If a file loaded from the host changes even slightly (including minor changes such as a timestamp change with the file contents left unmodified), then the Dagger cache will be invalidated. An extremely common source of invalidations occurs when loading the `.git` directory from the host filesystem, as that directory is constantly changing.
+
+  :::tip
+  To maximize cache re-use, it's important to use the include/exclude options for local directories to only include the files/directories needed for the pipeline. Excluding the `.git` directory is highly advisable.
+  :::
+
+- The host directory is loaded into the Dagger Engine; it's not a "bind mount". This means that any change you make to the loaded directory in your Dagger pipeline will not result in a change to the directory on the host.
+
+  :::warning
+  If you want the changes made to a loaded local directory inside a Dagger pipeline to be reflected on the host, it needs to be explictly exported to the host. However, this should be approached with great caution, since a mistake could result in files/directories being accidentally overwritten with undesired contents or unsaved work being deleted.
+  :::
+
+- Synchronization of a local directory happens once per Dagger "session" (in user-facing terms, once per `dagger.Connect` call in the Dagger SDKs). This means that if you load the local directory, then make changes to it on the host, those changes will not be reloaded within the context of a single Dagger client. Furthermore, due to lazy executions, the loading happens the first time the directory is used in a non-lazy operation.
```

Comment:
```suggestion
- Synchronization of a local directory happens once per Dagger client instance (in user-facing terms, once per `dagger.Connect` call in the Dagger SDKs). This means that if you load the local directory, then make changes to it on the host, those changes will not be reloaded within the context of a single Dagger client. Furthermore, due to lazy executions, the loading happens the first time the directory is used in a non-lazy operation.
```

I think we should avoid leaking the term "session" into here. It's a buildkit term technically and there's also changes underway that rework how sessions are used, so it's just better to avoid it entirely as to not accidentally teach users that it's the right term to use.